gcloud-java-resourcemanager/src/main/java/com/google/gcloud/resourcemanager/Project.java

```diff
@@ -0,0 +1,143 @@
+    this.resourceManager = checkNotNull(resourceManager);
+    this.info = checkNotNull(projectInfo);
+    this.policy = checkNotNull(policy);
+  }
+
+  /**
+   * Constructs a Project object that contains project and policy information loaded from the
+   * server.
+   *
+   * @return Project object containing the project's metadata and IAM policy
+   * @throws ResourceManagerException upon failure
+   */
+  public static Project load(ResourceManager resourceManager, String projectId) {
+    ProjectInfo projectInfo = resourceManager.get(projectId);
+    Policy policy = resourceManager.getIamPolicy(projectId);
+    return new Project(resourceManager, projectInfo, policy);
+  }
+
+  public ProjectInfo info() {
+    return info;
+  }
+
+  public Policy policy() {
+    return policy;
+  }
+
+  public ResourceManager resourceManager() {
+    return resourceManager;
+  }
+
+  /**
+   * Returns a Project object with updated project and policy information.
+   *
+   * @return Project object containing the project's updated metadata and IAM policy
+   * @throws ResourceManagerException upon failure
+   */
+  public Project reload() {
+    return new Project(
+        resourceManager, resourceManager.get(info.id()), resourceManager.getIamPolicy(info.id()));
+  }
+
+  /**
+   * Requests that this project be deleted. For an unspecified amount of time, this action can be
+   * undone by calling {@link #undelete}.
+   *
+   * @throws ResourceManagerException upon failure
+   */
+  public void delete() {
+    resourceManager.delete(info.id());
+  }
+
+  /**
+   * Requests that a project's lifecycle status be changed from {@code DELETE_REQUESTED} to
+   * {@code ACTIVE}.
+   *
+   * @throws ResourceManagerException upon failure
+   */
+  public void undelete() {
+    resourceManager.undelete(info.id());
+  }
+
+  /**
+   * Replaces the project metadata (not including the IAM policy) using the given ProjectInfo.
+   *
+   * @return Project object containing the project's updated metadata
+   * @throws ResourceManagerException upon failure
+   */
+  public Project replace(ProjectInfo projectInfo) {
+    return new Project(resourceManager, resourceManager.replace(checkNotNull(projectInfo)), policy);
+  }
+
+  /**
+   * Replaces the project's IAM policy using the given policy.
+   *
+   * @return Project object containing the project's updated IAM policy
+   * @throws ResourceManagerException upon failure
+   */
+  public Project replaceIamPolicy(Policy policy) {
+    return new Project(
+        resourceManager, info, resourceManager.replaceIamPolicy(info.id(), checkNotNull(policy)));
+  }
+
+  /**
+   * Returns whether the caller has the permissions specified in the parameters.
+   *
+   * @return List of booleans representing whether the user has the corresponding permission
+   *     provided as a parameter
+   * @throws ResourceManagerException upon failure
+   */
+  public List<Boolean> hasPermissions(Permission... permissions) {
```

Comment:
I also think this is a problem with binding Project to a Policy as the 2 `hasPermissions` method do not consider the attached `Policy`.